### ISSUE TITLE
Option to ignore deprecated versions

### ISSUE THREAD

**Issue by nseni** [raineorshine/npm-check-updates#624]
- [x] I have read the list of [known issues](https://github.com/tjunnone/npm-check-updates#known-issues) before filing this issue.
- [x] I have searched for [similiar issues](https://github.com/tjunnone/npm-check-updates/issues) before filing this issue.

* node version: 12.13.0
* npm version: 6.12.0
* npm-check-updates version: 4.0.1

---------------------------
Sometimes package authors release a new version by mistake and they mark such release as deprecated. For example: https://www.npmjs.com/package/primeng?activeTab=versions

![image](https://user-images.githubusercontent.com/47851128/73525969-41fe0f00-4421-11ea-8f4a-20e26841d19f.png)

But npm-check-updates still suggests to update to the latest version:
![image](https://user-images.githubusercontent.com/47851128/73526184-b20c9500-4421-11ea-8e67-9642b0476fae.png)

Can anything be done about it?

**Comment by raineorshine** [comment]
The deprecation status is retrieved with `pacote`:

```
$ pacote packument primeng | grep deprecated
      "deprecated": "Invalid configuration, use primeng-lts@8.1.3 instead"
      "deprecated": "Invalid configuration, use primeng-lts@8.1.x instead"
```

Which is what `npm-check-updates` [uses to fetch package data](https://github.com/tjunnone/npm-check-updates/blob/master/lib/package-managers/npm.js#L73).

So, yes, deprecated versions could be ignored if the `deprecated` field was checked.

**Comment by raineorshine** [comment]
It's probably a good practice for package authors to update their `latest` tag to a non-deprecated version.

Still, I don't see any downside to `ncu` printing a message and filtering deprecated versions out if one is detected.

**Comment by Jaimeloeuf** [comment]
Hey @raineorshine thanks for the explanation for this, but is there any updates on this? Since <https://github.com/jakubroztocil/rrule/issues/417> issues like this still happen fairly frequently, it would be great if this package ignores the deprecated versions by default!

**Comment by raineorshine** [comment]
Fixed and published in [v9.0.0](https://github.com/raineorshine/npm-check-updates/releases/tag/v9.0.0).
